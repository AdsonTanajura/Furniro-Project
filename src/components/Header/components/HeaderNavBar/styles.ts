import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const MenuList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  width: 100vw;
  max-width: 430px;

  @media (max-width: 780px) {
    flex-direction: column;
  }
`;

export const MenuLink = styled(Link)`
  text-decoration: none;
  font-size: 16px;
  color: #000000;
  transition: border-bottom 0.2s;

  &:hover {
    border-bottom: 2px #000000 solid;
  }
`;
