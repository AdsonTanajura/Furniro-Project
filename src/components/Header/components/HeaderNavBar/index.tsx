import { MenuLink, MenuList } from './styles';

const HeaderNavBar = () => {
  return (
    <nav>
      <MenuList>
        <li>
          <MenuLink to={'/home'}>Home</MenuLink>
        </li>
        <li>
          <MenuLink to={'/home/shop'}>Shop</MenuLink>
        </li>
        <li>
          <MenuLink to={'/a'}>About</MenuLink>
        </li>
        <li>
          <MenuLink to={'/home/contact'}>Contact</MenuLink>
        </li>
      </MenuList>
    </nav>
  );
};

export default HeaderNavBar;
