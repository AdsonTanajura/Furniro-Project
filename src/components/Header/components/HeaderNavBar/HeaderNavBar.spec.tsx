import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import HeaderNavBar from '.';

describe('<HeaderNavBar />', () => {
  it('should render all the elements of the component', () => {
    const { getAllByRole, container } = render(
      <MemoryRouter>
        <HeaderNavBar />
      </MemoryRouter>
    );

    const ulElements = getAllByRole('list');
    const liElements = getAllByRole('listitem');
    const navElement = container.firstChild;
    const linkElements = getAllByRole('link');

    expect(linkElements.length).toBeGreaterThanOrEqual(4);
    expect(linkElements.length).toBeLessThanOrEqual(4);
    linkElements.forEach((linkElement) =>
      expect(linkElement).toBeInTheDocument()
    );
    linkElements.forEach((linkElement) => expect(linkElement).toBeVisible());

    expect(liElements.length).toBeGreaterThanOrEqual(4);
    expect(liElements.length).toBeLessThanOrEqual(4);
    liElements.forEach((liElement) => expect(liElement).toBeInTheDocument());
    liElements.forEach((liElement) => expect(liElement).toBeVisible());

    expect(ulElements.length).toBeGreaterThanOrEqual(1);
    ulElements.forEach((ulElement) => expect(ulElement).toBeInTheDocument());
    ulElements.forEach((ulElement) => expect(ulElement).toBeVisible());

    expect(navElement).toBeInTheDocument();
    expect(navElement).toBeVisible();
  });

  it('should render the correct links', () => {
    const { getByRole } = render(
      <MemoryRouter>
        <HeaderNavBar />
      </MemoryRouter>
    );

    expect(getByRole('link', { name: /home/i })).toHaveAttribute(
      'href',
      '/home'
    );
    expect(getByRole('link', { name: /shop/i })).toHaveAttribute(
      'href',
      '/home/shop'
    );
    expect(getByRole('link', { name: /about/i })).toHaveAttribute('href', '/a');
    expect(getByRole('link', { name: /contact/i })).toHaveAttribute(
      'href',
      '/home/contact'
    );
  });

  it('should navigate to the correct page by clicking on the links', async () => {
    const { getByRole, findByTestId } = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<HeaderNavBar />} />
          <Route
            path="/home"
            element={<div data-testid="home-page">home Page</div>}
          />
        </Routes>
      </MemoryRouter>
    );

    const user = userEvent.setup();
    const homeLink = getByRole('link', { name: /home/i });

    await user.click(homeLink);

    expect(await findByTestId('home-page')).toBeInTheDocument();
  });
});
