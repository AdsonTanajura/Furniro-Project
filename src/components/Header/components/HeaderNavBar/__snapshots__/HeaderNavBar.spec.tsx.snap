// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderNavBar /> should navigate to the correct page by clicking on the links 1`] = `
@media (max-width: 780px) {

}

<div
  data-testid="home-page"
>
  home Page
</div>
`;

exports[`<HeaderNavBar /> should render all the elements of the component 1`] = `
.c0 {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  width: 100vw;
  max-width: 430px;
}

.c1 {
  text-decoration: none;
  font-size: 16px;
  color: #000000;
  transition: border-bottom 0.2s;
}

.c1:hover {
  border-bottom: 2px #000000 solid;
}

@media (max-width: 780px) {
  .c0 {
    flex-direction: column;
  }
}

<nav>
  <ul
    class="c0"
  >
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home/shop"
      >
        Shop
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/a"
      >
        About
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<HeaderNavBar /> should render the correct links 1`] = `
.c0 {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  width: 100vw;
  max-width: 430px;
}

.c1 {
  text-decoration: none;
  font-size: 16px;
  color: #000000;
  transition: border-bottom 0.2s;
}

.c1:hover {
  border-bottom: 2px #000000 solid;
}

@media (max-width: 780px) {
  .c0 {
    flex-direction: column;
  }
}

<nav>
  <ul
    class="c0"
  >
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home/shop"
      >
        Shop
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/a"
      >
        About
      </a>
    </li>
    <li>
      <a
        class="c1"
        data-discover="true"
        href="/home/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
